acorn/src/expression.js

```diff
@@ -548,11 +565,8 @@ pp.parseNew = function() {
     return this.finishNode(node, "MetaProperty")
   }
   let startPos = this.start, startLoc = this.startLoc
-  node.callee = this.parseSubscripts(this.parseExprAtom(), startPos, startLoc, true)
-  if (this.options.ecmaVersion > 10 && node.callee.type === "Import") {
-    this.raise(node.callee.start, "Cannot use new with import(...)")
-  }
-  if (this.eat(tt.parenL)) node.arguments = this.parseExprList(tt.parenR, this.options.ecmaVersion >= 8 && node.callee.type !== "Import", false)
+  node.callee = this.parseSubscripts(this.parseExprAtom(null, true), startPos, startLoc, true)
```

Comment:
What is the advantage of adding a new argument to `parseExprAtom` over the way the check worked before?

Reply:
Because 

1. `new (import(s))` is the valid syntax, so the existing logic is wrong.
1. `new import.meta` is the valid syntax, in the future. If acorn throws the parse error there, we make `import.meta` plugin relatively hard.

Reply:
Hm. I will replace the new argument by another way.